schema/connection.go

```diff
@@ -0,0 +1,18 @@
+package schema
+
+// Connection is a dummy validator to define a weak connection to another
+// schema. The query.Projection will treat this validator as an external
+// resource, and generate a sub-request to fetch the sub-payload.
+type Connection struct {
+	Path string
+}
+
+func (v *Connection) Compile(rc ReferenceChecker) (err error) {
+	// Nothing to compile, implemented to force Connection on pointer.
```

Comment:
This is hacky, just move Validate on pointer as that will do the same, but compile time.